### DISCUSSION TITLE
Automatically released `binaries`

### DISCUSSION THREAD

**Discussion by gwbres** [nav-solutions/rinex#57] [General]
@larsnaesbye,

For information, I had to introduce the `SRC_DIR` variable to the release script. "rinex" is a "cargo workspace" and this is how we define the subdirectories. In this scenario, we will get the "rinex-cli" binary generated. Ideally, we should come up with a final script where we also build the "crx2rnx" and "rnx2crx" binaries

- [x] the Windows binary works, it's probably the most mandatory
- [ ] then MacOS does compile but it seems the publication is in failure.
And I think the problem is related to the rust builder.. It's like the previous "windows" job created a local file, and we cannot overwrite it. But I might be wrong..
"cp: can't create directory '/github/workspace/output/rinex-cli': File exists"

- [ ] the linux job is in failure due to the rinex-cli "libfontconfig-dev" dependency.
The dependency is met. But `pkg-config` does not locate it. 
We simply need to specify the library location, either by an env. variable or via `pkg-config`. I should be able to fix that

**Comment by gwbres** [comment]
I'll leave it there for the moment.. The current status is:

- all three jobs exist, failures are disregarded, I split it into 3 jobs so whatever passes gets published
- Binaries are available for Windows, I'll mention that in the docs
- There is an issue when it's about to publish in the Mac/Docker context, and it only exists in that context
- The Linux-musl job is in failure. We need the `PKG_CONFIG_PATH` variable to point to `usr/lib/x86-64_linux-gnu/pkgconfig/fontconfig.pc` or a very similar location.

**Comment by larsnaesbye** [comment]
That’s cool. I had my share of fighting with it too, and we can add builds over time.

**Reply by gwbres** [reply]
> That’s cool. I had my share of fighting with it too, and we can add builds over time.

Hello Lars,
FYI, to design your release scripts, set a `pull_request` trigger temporarily, and work in a PR.
I'm improving the rinex one, so the binary has a more meaningful name

**Reply by larsnaesbye** [reply]
Good idea. That way you don't need to make releases and delete them.

**Comment by larsnaesbye** [comment]
I did manage to build a macOS version over at my project : https://github.com/larsnaesbye/Tempus_port/actions/runs/3576992909/jobs/6015463510

**Reply by larsnaesbye** [reply]
OK, that did not work with this project. Tried to make a copy and reuse my Actions - failed.

**Reply by gwbres** [reply]
> OK, that did not work with this project. Tried to make a copy and reuse my Actions - failed.

this is weird. The action script look similar. The only difference I would say, is the base of this project being a "cargo workspace".

**Comment by larsnaesbye** [comment]
It seems like [this tool](https://github.com/houseabsolute/actions-rust-cross) might do the job better. The author uses it to build their own code quality tool.
